Source/Core/VideoCommon/VertexLoaderX64.cpp

```diff
@@ -98,30 +98,111 @@ int VertexLoaderX64::ReadVertex(OpArg data, u64 attribute, int format, int count
 
 	int elem_size = 1 << (format / 2);
 	int load_bytes = elem_size * count_in;
-	if (load_bytes > 8)
-		MOVDQU(coords, data);
-	else if (load_bytes > 4)
-		MOVQ_xmm(coords, data);
-	else
-		MOVD_xmm(coords, data);
+	OpArg dest = MDisp(dst_reg, m_dst_ofs);
+
+	native_format->components = count_out;
+	native_format->enable = true;
+	native_format->offset = m_dst_ofs;
+	native_format->type = VAR_FLOAT;
+	native_format->integer = false;
 
-	PSHUFB(coords, M(&shuffle_lut[format][count_in - 1]));
+	m_dst_ofs += sizeof(float) * count_out;
 
-	if (format != FORMAT_FLOAT)
+	if (attribute == DIRECT)
+		m_src_ofs += load_bytes;
+
+	if (cpu_info.bSSSE3)
 	{
-		// Sign extend
+		if (load_bytes > 8)
+			MOVDQU(coords, data);
+		else if (load_bytes > 4)
+			MOVQ_xmm(coords, data);
+		else
+			MOVD_xmm(coords, data);
+
+		PSHUFB(coords, M(&shuffle_lut[format][count_in - 1]));
+
+		// Sign-extend.
 		if (format == FORMAT_BYTE)
 			PSRAD(coords, 24);
 		if (format == FORMAT_SHORT)
 			PSRAD(coords, 16);
+	}
+	else
+	{
+		// SSE2
+		switch (format)
+		{
+		case FORMAT_UBYTE:
+			MOVD_xmm(coords, data);
+			PXOR(XMM1, R(XMM1));
+			PUNPCKLBW(coords, R(XMM1));
+			PUNPCKLWD(coords, R(XMM1));
+			break;
+		case FORMAT_BYTE:
+			MOVD_xmm(coords, data);
+			PUNPCKLBW(coords, R(coords));
+			PUNPCKLWD(coords, R(coords));
+			PSRAD(coords, 24);
+			break;
+		case FORMAT_USHORT:
+		case FORMAT_SHORT:
+			switch (count_in)
+			{
+			case 1:
+				LoadAndSwap(16, scratch3, data);
+				MOVD_xmm(coords, R(scratch3));
+				break;
+			case 2:
+				LoadAndSwap(32, scratch3, data);
+				MOVD_xmm(coords, R(scratch3));
+				PSHUFLW(coords, R(coords), 0xC5); // ABCD EFGH -> ABCD EHGG
+				break;
+			case 3:
+				LoadAndSwap(64, scratch3, data);
+				MOVQ_xmm(coords, R(scratch3));
+				PSHUFLW(coords, R(coords), 0xE7); // ABCD EFGH -> ABCD EFGE
+				PUNPCKLQDQ(coords, R(coords));
+				PSHUFHW(coords, R(coords), 0xE5); // ABCD EFGH -> ABCC EFGH
```

Comment:
The comment seems to just describe the operation itself; maybe have the comments track the positions of the input values to make it more clear?  e.g. ...

// A B C _ _ _ _ _
// A B C _ A B C _ (punpcklqdq)
// A A B B A B C _ (pshuflw)
// A A B B C C _ _ (pshufhw)